src/extension/ddappsec.c

```diff
@@ -310,6 +311,31 @@ static ZEND_INI_MH(_on_update_appsec_enabled_on_cli)
     return SUCCESS;
 }
 
+static ZEND_INI_MH(_on_update_unsigned)
+{
+    ZEND_INI_MH_UNUSED();
+
+    char *endptr = NULL;
+    long ini_value = strtol(ZSTR_VAL(new_value), &endptr, 10);
+
+    // If there is any error parsing, don't set the value.
+    if (endptr == ZSTR_VAL(new_value) || *endptr != '\0') {
+        return SUCCESS;
```

Comment:
should return FAILURE, otherwise it's misleading to the user (php.ini will show the bad value)